lib/shared-class.js

```diff
@@ -212,9 +214,58 @@ SharedClass.prototype.find = function(fn, isStatic) {
  */
 
 SharedClass.prototype.disableMethod = function(fn, isStatic) {
-  var disableMethods = this._disabledMethods;
+  deprecated('SharedClass.prototype.disableMethod is deprecated. ' +
+    'Use SharedClass.prototype.disableMethodByName instead.');
   var key = this.getKeyFromMethodNameAndTarget(fn, isStatic);
-  disableMethods[key] = true;
+  this._disableSharedMethod(key);
+};
+
+/**
+ * Internal implementation to find a sharedMethod.
+ *
+ * @param {String|Function} fn The function or method name
+ * @param {Boolean} [isStatic] Required if `fn` is a `String`.
+ * Only find a static method with the given name.
+ * @returns {SharedMethod}
+ */
+
+SharedClass.prototype._findSharedMethod = function(fn, isStatic) {
+  var methods = this.methods();
+  return find(methods, fn, isStatic);
+};
+
+/**
+ * Internal implementation to disable a sharedMethod.
+ *
+ * @param {String} methodOrFnName The function or method name
```

Comment:
I would expect that in this private function, the argument must be always a string (method name):
1) `this._disabledMethods` is indexed by strings IIRC 
2) the deprecated public method `disableMethods` should take care of conversion from function to method name.